crates/ruff_python_formatter/src/other/f_string.rs

```diff
@@ -1,12 +1,48 @@
+use ruff_formatter::FormatRuleWithOptions;
 use ruff_python_ast::FString;
+use ruff_text_size::Ranged;
 
 use crate::prelude::*;
+use crate::string::{StringContext, StringPart};
 
 #[derive(Default)]
-pub struct FormatFString;
+pub struct FormatFString {
+    context: StringContext,
```

Comment:
This is beneficial but not completely in the way you expected mainly because of my comment above (https://github.com/astral-sh/ruff/pull/9058#discussion_r1425727018).

Reply:
I've made this change and updated the `generate.py` script to ignore generating impls for `FString` node. Do you think it would be useful to move the file from `other/f_string.rs` to `string/f_string.rs` crate?